### ISSUE TITLE
[Feature Request] A11y Role differentiation

### ISSUE THREAD

**Issue by jwoo-msft** [microsoft/AdaptiveCards#8138]
### Problem Statement

when authors create card with action.openurl selection action, adaptivecard renderer has no good way of knowing if the action should be button or link.

if the action looks like a button, its a11y role should be button regardless of what action the select action performs.

renderers cannot tell if the select action looks like a button or not. 

We have several related a11y issues.

### Proposed solution

a new property `role` is added to distinguish between button and link with default role being button.

select action work as link only in limited situations such as select action in TextBlock or RichTextBlocks. Most of cases select actions are enclosed in a container, and they tend to look like buttons; however, we can't completely rule out a case where the container looks like a link. 

giving options to card authors to explicitly assign the role can solve all of the use cases.

less explicitly but doable in a sense that addresses most of the issue is assigning link role only to select actions that are associated with with text.

### Alternatives or Workarounds

authors may use action.submit or action.execute, but it degrades experiences. with action.openurl, users could open app immediately. with the workarounds, users have to execute additional steps to open app.

**Comment by anna-dingler** [comment]
Closing this issue since the feature has been completed.
